Generate.py

```diff
@@ -256,18 +333,217 @@ def get_choice_legacy(option, root, value=None) -> Any:
     raise RuntimeError(f"All options specified in \"{option}\" are weighted as zero.")
 
 
-def get_choice(option, root, value=None) -> Any:
-    if option not in root:
-        return value
-    if type(root[option]) is list:
-        return random.choices(root[option])[0]
-    if type(root[option]) is not dict:
-        return root[option]
-    if not root[option]:
-        return value
-    if any(root[option].values()):
-        return random.choices(list(root[option].keys()), weights=list(map(int, root[option].values())))[0]
-    raise RuntimeError(f"All options specified in \"{option}\" are weighted as zero.")
+BinOp = Literal["+", "-", "*", "/"]
+
+
+_bops: Mapping[BinOp, Callable[[Union[float, int], Union[float,int]], Union[float, int]]] = {
+    "+": lambda a, b: a + b,
```

Comment:
The PR author is already aware of this, but this is a note to other reviewers.

This isn't a good place for this math parsing code, but it's not a lot worse than any other place that's available.

It would be good to put it in its own module as a sub-module to `Utils` https://github.com/ArchipelagoMW/Archipelago/pull/4032